Runtime/ui/painting/txt/mesh_generator.cs

```diff
@@ -225,58 +223,61 @@ class TextBlobMesh : PoolObject {
                     uvCoord.Add(uvRect.topRight.toVector());
                     uvCoord.Add(uvRect.topLeft.toVector());
 
-                    if(char.IsHighSurrogate(a)) i++;
+                    if (char.IsHighSurrogate(a)) {
+                        i++;
+                    }
                 }
-                
+
                 uiMeshMesh meshMesh = uiMeshMesh.create(null, vert, tri, uvCoord);
-                
+
                 if (_meshes.ContainsKey(key)) {
                     ObjectPool<MeshInfo>.release(_meshes[key]);
                     _meshes.Remove(key);
                 }
+
                 _meshes[key] = MeshInfo.create(key, meshMesh, 0);
 
                 this._mesh = meshMesh.transform(this.matrix);
                 return this._mesh;
             }
-            
+
             var length = this.textBlob.textSize;
             var fontSizeToLoad = Mathf.CeilToInt(style.UnityFontSize * this.scale);
 
             var vertices = ObjectPool<uiList<Vector3>>.alloc();
             vertices.SetCapacity(length * 4);
-            
+
             var triangles = ObjectPool<uiList<int>>.alloc();
             triangles.SetCapacity(length * 6);
 
             var uv = ObjectPool<uiList<Vector2>>.alloc();
             uv.SetCapacity(length * 4);
-            
+
             for (int charIndex = 0; charIndex < length; ++charIndex) {
                 var ch = text[charIndex + this.textBlob.textOffset];
                 // first char as origin for mesh position 
-                var position = this.textBlob.positions[charIndex];
+                var position = this.textBlob.getPosition(charIndex);
                 if (LayoutUtils.isWordSpace(ch) || LayoutUtils.isLineEndSpace(ch) || ch == '\t') {
                     continue;
                 }
 
                 if (fontSizeToLoad == 0) {
                     continue;
                 }
-                
+
                 font.getGlyphInfo(ch, out var glyphInfo, fontSizeToLoad, style.UnityFontStyle);
-                
+
                 var minX = glyphInfo.minX / this.scale;
                 var maxX = glyphInfo.maxX / this.scale;
                 var minY = -glyphInfo.maxY / this.scale;
                 var maxY = -glyphInfo.minY / this.scale;
 
                 var baseIndex = vertices.Count;
 
-                vertices.Add(new Vector3((position.x + minX), (position.y + minY), 0));
-                vertices.Add(new Vector3((position.x + maxX), (position.y + minY), 0));
-                vertices.Add(new Vector3((position.x + maxX), (position.y + maxY), 0));
-                vertices.Add(new Vector3((position.x + minX), (position.y + maxY), 0));
+                vertices.Add(new Vector3(position + minX, minY, 0));
```

Comment:
position => positionX, to make it clear that this "position" is not a real (x,y) like position but x only ?

Reply:
Same as above.